pkg/multiplexer/multiplexer_test.go

```diff
@@ -163,10 +164,13 @@ func startServer() *http.Server {
 		}
 	})
 
-	testServer := &http.Server{Addr: "localhost:9990", Handler: instance}
+	listener, err := net.Listen("tcp", "localhost:9990")
+	assert.NoError(t, err)
+
+	testServer := &http.Server{Handler: instance}
```

Comment:
Not entirely related to this change, but I'm hoping this fixes our tests similar to how the Voltron tests were fixed.